xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -1701,9 +1702,10 @@ glusterd_volinfo_find(const char *volname, glusterd_volinfo_t **volinfo)
     priv = this->private;
     GF_ASSERT(priv);
 
+    hash = gf_dm_hashfn(volname, strlen(volname));
     cds_list_for_each_entry(tmp_volinfo, &priv->volumes, vol_list)
     {
-        if (!strcmp(tmp_volinfo->volname, volname)) {
+        if (tmp_volinfo->volname_hash == hash) {
```

Comment:
This is not enough. We also need to compare the full volname. Hashes may collide.